source/client.c

```diff
@@ -488,16 +486,23 @@ int aws_mqtt_client_connection_connect(
         aws_string_destroy(connection->host_name);
     }
 
-    connection->host_name = aws_string_new_from_array(connection->allocator, host_name->ptr, host_name->len);
-    connection->port = port;
-    connection->socket_options = *socket_options;
+    connection->host_name = aws_string_new_from_array(
+        connection->allocator, connection_options->host_name->ptr, connection_options->host_name->len);
+    connection->port = connection_options->port;
+    connection->socket_options = *connection_options->socket_options;
     connection->state = AWS_MQTT_CLIENT_STATE_CONNECTING;
-    connection->clean_session = clean_session;
-    connection->keep_alive_time = keep_alive_time;
+    connection->clean_session = connection_options->clean_session;
+    connection->keep_alive_time = connection_options->keep_alive_time_secs;
+    if (!connection_options->request_timeout_ms) {
+        connection->request_timeout_ns = s_default_request_timeout_ns;
+    } else {
+        connection->request_timeout_ns = aws_timestamp_convert(
+            (uint64_t)connection_options->request_timeout_ms, AWS_TIMESTAMP_MILLIS, AWS_TIMESTAMP_NANOS, NULL);
+    }
 
     /* Cheat and set the tls_options host_name to our copy if they're the same */
-    if (tls_options) {
-        connection->tls_options = *tls_options;
+    if (connection_options->tls_options) {
+        connection->tls_options = *connection_options->tls_options;
```

Comment:
Doesn't this need to use aws_tls_connection_options_copy?

Reply:
no, it gets copied by the bootstrap.